docs/how-tos/how-to-add-advanced-search.mdx

```diff
@@ -0,0 +1,143 @@
+---
+sidebar_position: 7
+---
+
+# How to add advanced search to Solidus
+
+In this guide, we'll see how to improve the default search capabilities
+offered with Solidus by default extending its potential with features like
+search autocomplete with suggestions, typo tolerance.
+
+This how-to will use a gem called [Searchkick][searchkick-github], which
+is one of most popular choices in the Rails ecosystem.
+
+## Installing Searchkick
+
+:::caution
+
+Searchkick supports both Elasticsearch and OpenSearch. This guide
+section covers the installation with Elasticsearch. Please, refer to
+[Searchkick README][searchkick-github] for more installation options.
+
+:::
+
+:::info
+
+Be sure you have Elasticsearch up and running. Please, follow the
+official [installation instructions][elasticsearch-install-guide],
+or if you are a MacOS and Homebrew user, just run:
+
+```
+brew install elastic/tap/elasticsearch-full
+brew services start elasticsearch-full
+```
+
+:::
+
+Let's get started by adding the necessary gems to our Gemfile:
+
+```
+bundle add searchkick elasticsearch
+```
+
+## Product Indexing Configuration
+
+
+We can now customize our Product model to tell it how its information
+should be indexed. We can [customize][override-core] the `Spree::Product`
+core model:
+
+```ruby title="app/overrides/my_store/spree/product/add_searchkick.rb"
+module MyStore
+  module Spree
+    module Product
+      module AddSearchkick
+
+
+        # TODO: add content complete
+
+
+        ::Spree::Product.prepend self
+      end
+    end
+  end
+end
+```
+
+Now it's time for our first indexing.
+
+```ruby title="bin/rails c"
+> Spree::Product.reindex
+```
+
+or
+
+```sh
+$ rake searchkick:reindex CLASS=Spree::Product
```

Comment:
`rake` → `bin/rails`

This correctly loads rake from bundler and is the official way of calling it in rails.